### PR TITLE
Revert "add componentstatus metrics"

### PR DESCRIPTION
**Pull request by andyxning**
This reverts commit 5452f063935185d32c2fffcdbcf1f3decf1db3da.

As for now `componentstatuses` does not support `watch` method, we can not watch on it.  This should fix [219 comment](https://github.com/kubernetes/kube-state-metrics/pull/219#issuecomment-351824814).

ping @zouyee 
/cc @brancz

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kube-state-metrics/317)

### CODE REVIEW

**Comment by zouyee** [comment]
https://github.com/kubernetes/client-go/blob/v5.0.0/informers/core/v1/componentstatus.go#L49

File: main.go [hunk 1]
```diff
@@ -66,7 +65,6 @@ var (
 		"horizontalpodautoscalers": struct{}{},
 	}
 	availableCollectors = map[string]func(registry prometheus.Registerer, kubeClient clientset.Interface, namespace string){
-		"componentstatuses":        collectors.RegisterComponentStatusCollector,
```
Comment (zouyee) [inline main.go]: `componentstatuses` options need to be removed from `--collectors`

Comment (andyxning) [reply main.go]: @zouyee The default value for `--collectors` is of `defaultCollectors` keys which has `componentstatuses` removed.

**Comment by brancz** [comment]
I don't think we have to revert, let's just fix the problem I described in the comment on the original PR.

**Comment by brancz** [comment]
You were right, sorry, I didn't quite understand at first.

**Pull request merged**